### PR TITLE
Bump github.com/goreleaser/goreleaser from 1.25.1 to 1.26.0

### PR DESCRIPTION
**Pull request by jpkrohling**
Bumps [github.com/goreleaser/goreleaser](https://github.com/goreleaser/goreleaser) from 1.25.1 to 1.26.0.
- [Release notes](https://github.com/goreleaser/goreleaser/releases)
- [Changelog](https://github.com/goreleaser/goreleaser/blob/main/.goreleaser.yaml)
- [Commits](https://github.com/goreleaser/goreleaser/compare/v1.25.1...v1.26.0)

---
updated-dependencies:
- dependency-name: github.com/goreleaser/goreleaser
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

### CODE REVIEW

**Comment by jpkrohling** [comment]
Replaces https://github.com/open-telemetry/opentelemetry-collector-releases/pull/550

**Comment by jpkrohling** [comment]
@TylerHelmuth , are you able to review this one?

File: go.mod [hunk 1]
```diff
@@ -1,11 +1,11 @@
 module github.com/open-telemetry/opentelemetry-collector-releases
 
-go 1.21
+go 1.22
```
Comment (songy23) [inline go.mod:3]: I think we want to keep 1.21 for now?

File: go.mod [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 
-toolchain go1.21.7
+toolchain go1.22.3
```
Comment (songy23) [inline go.mod:5]: Same here

**Comment by jpkrohling** [comment]
Actually, it looks like @pjanotti's change adding MSI incorporated the version bump already. I think we are good. Thank for the review though!

**Pull request closed without merging**